solutions/standard/variables.tf

```diff
@@ -0,0 +1,133 @@
+##############################################################################
+# Input Variables
+##############################################################################
+
+variable "ibmcloud_api_key" {
+  type        = string
+  description = "The IBM Cloud API key to deploy resources."
+  sensitive   = true
+}
+
+variable "provider_visibility" {
+  description = "Set the visibility value for the IBM terraform provider. Supported values are `public`, `private`, `public-and-private`. [Learn more](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/guides/custom-service-endpoints)."
+  type        = string
+  default     = "private"
+
+  validation {
+    condition     = contains(["public", "private", "public-and-private"], var.provider_visibility)
+    error_message = "Invalid visibility option. Allowed values are 'public', 'private', or 'public-and-private'."
+  }
+}
+
+variable "use_existing_resource_group" {
+  type        = bool
+  description = "Whether to use an existing resource group."
+  default     = false
+}
+
+variable "resource_group_name" {
+  type        = string
+  description = "The name of a new or an existing resource group to provision the watsonx.data in. If a prefix input variable is specified, the prefix is added to the name in the `<prefix>-<name>` format."
+}
+
+variable "prefix" {
+  type        = string
+  description = "(Optional) Prefix to add to all resources created by this solution. To not use any prefix value, you can set this value to `null` or an empty string."
+  default     = "wx-data"
+}
+
+variable "name" {
+  type        = string
+  description = "The name of the watsonx.data instance. If a prefix input variable is specified, the prefix is added to the name in the `<prefix>-<name>` format."
+  default     = "dev"
+}
+
+variable "region" {
+  description = "The region where you want to deploy your instance."
+  type        = string
+  default     = "us-south"
+}
+
+variable "plan" {
+  type        = string
+  description = "The plan that is required to provision the watsonx.data instance. Possible values are: `lite` , `lakehouse-enterprise` or `lakehouse-enterprise-mcsp` only for `au-syd` region. [Learn more](https://cloud.ibm.com/docs/watsonxdata?topic=watsonxdata-getting-started_1)."
+  default     = "lakehouse-enterprise"
+  validation {
+    condition = anytrue([
+      var.region == "au-syd" && var.plan == "lakehouse-enterprise-mcsp",
+      var.region != "au-syd" && contains(["lakehouse-enterprise", "lite"], var.plan)
+    ])
+    error_message = "If the region is 'au-syd', the plan must be 'lakehouse-enterprise-mcsp'. For other supported regions, the plan must be 'lakehouse-enterprise' or 'lite'."
```

Comment:
It means we cant create lite plan instance in au-syd. is it so?

Reply:
Hmm...that's a good catch. I am correcting this.

`lite` plan + any supported region - always provision
`lakehouse-enterprise` + all supported regions except au-syd - always provision
`lakehouse-enterprise-mcsp` + only region `au-syd` - will be provisioned.

Reply:
Corrected this.